shared/src/validation-schemas/sign-up/sign-up.schema.ts

```diff
@@ -0,0 +1,15 @@
+import * as Joi from 'joi';
+import { UserCreatePayload } from '~/common/types/types';
+
+const SignupSchema = Joi.object<UserCreatePayload>({
+  firstName: Joi.string().min(1).max(20).required(),
+  lastName: Joi.string().min(1).max(20).required(),
```

Comment:
```suggestion
  lastName: Joi.string().min(1).max(SignUpValidationRule.LAST_NAME_MAX_LENGTH).required(),
```
Create enum for:
rules - SignUpValidationRule
messages: SignUpValidationMessage
in the `shared/src/common/enums/validation/`